src/pal/src/configure.cmake

```diff
@@ -73,8 +73,33 @@ check_function_exists(directio HAVE_DIRECTIO)
 check_function_exists(semget HAS_SYSV_SEMAPHORES)
 check_function_exists(pthread_mutex_init HAS_PTHREAD_MUTEXES)
 check_function_exists(ttrace HAVE_TTRACE)
+if(CMAKE_SYSTEM_NAME STREQUAL Linux)
+set(CMAKE_REQUIRED_LIBRARIES unwind unwind-generic)
+check_cxx_source_compiles("
+#include <libunwind.h>
+
+int main(int argc, char **argv) {
+  unw_cursor_t cursor;
+  unw_save_loc_t saveLoc;
+  int reg = UNW_REG_IP;
+  unw_get_save_loc(&cursor, reg, &saveLoc);
+
+  return 0;
+}" HAVE_UNW_GET_SAVE_LOC)
+check_cxx_source_compiles("
+#include <libunwind.h>
+
+int main(int argc, char **argv) {
+  unw_addr_space_t as;
+  unw_get_accessors(as);
+
+  return 0;
+}" HAVE_UNW_GET_ACCESSORS)
+set(CMAKE_REQUIRED_LIBRARIES)
+else(CMAKE_SYSTEM_NAME STREQUAL Linux)
 check_function_exists(unw_get_save_loc HAVE_UNW_GET_SAVE_LOC)
```

Comment:
Not the OSX - that is using the LLVM libunwind, but since it doesn't have the unw_get_save_loc and unw_get_accessors, the effect of the check_source_compiles vs check_function_exists would be the same.